### PR TITLE
Controllers specs completed

### PR DESCRIPTION
**Pull request by ometman**
I have completed specs for the controllers, on branch controller-specs:

- Added RSpec as a gem to the project.
- Created a Request spec file for all controllers.
- For each action:
    - Response status was checked and correct.
    - A correct template was checked and rendered.
    - The response body includes correct placeholder text.
 - Linters passed.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,12 +1,12 @@
 class UsersController < ApplicationController
   def index
     @users = User.all
-    render json: @users
+    render 'index'
```
Comment (LionRouge1) [inline app/controllers/users_controller.rb:4]: [OPTIONAL]
- [ ] Kindly note that rails automatically render the view that have the same name as the action in the controller. Therefore there is no need to add the `render 'index'` to the `index` action, rails will render `app/views/users/index.html.erb` by default for the `index` action in the users controller.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
   end
 
   def show
     @user = User.find(params[:id])
     @user_posts = @user.posts
-    render json: @user_posts
+    render 'show'
```
Comment (LionRouge1) [inline app/controllers/users_controller.rb:10]: [OPTIONAL]
- [ ] Same here as pointed in the previous comment :point_up_2: rails will render the `app/views/users/show.html.erb` by default for the `show` action in the users controller. Therefore I suggest you remove this `render 'show'`

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -4,11 +4,11 @@ class PostsController < ApplicationController
 
   def index
     @posts = Post.all
-    render json: @posts
+    render 'index'
```
Comment (LionRouge1) [inline app/controllers/posts_controller.rb:7]: [OPTIONAL]
- [ ] Kindly note that rails automatically render the view that have the same name as the action in the controller. Therefore there is no need to add the `render 'index'` to the `index` action, rails will render `app/views/posts/index.html.erb` by default for the `index` action in the posts controller.

File: app/controllers/posts_controller.rb [hunk 4]
```diff
@@ -4,11 +4,11 @@ class PostsController < ApplicationController
...
   end
 
   def show
-    render json: @post
+    render 'show'
```
Comment (LionRouge1) [inline app/controllers/posts_controller.rb:11]: [OPTIONAL]
- [ ] Same here as pointed in the previous comment :point_up_2: rails will render the `app/views/posts/show.html.erb` by default for the `show` action in the posts controller. Therefore I suggest you remove this `render 'show'`

**Review by LionRouge1 — APPROVED** [review]
Hi @ometman, 

# STATUS: APPROVED! 🥇 💯 
Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:

done

## Highlights
- Great job, your project is error free :+1: 
- Nice work you meet all the requirements :1st_place_medal: 
- All linters passed :heavy_check_mark: 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 4 above)
Comment (ometman) [reply app/controllers/posts_controller.rb:11]: 👍

File: app/controllers/posts_controller.rb (hunk 3 above)
Comment (ometman) [reply app/controllers/posts_controller.rb:7]: 👍

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (ometman) [reply app/controllers/users_controller.rb:10]: 👍

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (ometman) [reply app/controllers/users_controller.rb:4]: 👍

**Pull request merged**